flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1195,9 +1184,22 @@ private ApplicationReport startAppMaster(
         yarnClient.submitApplication(appContext);
 
         LOG.info("Waiting for the cluster to be allocated");
+
+        waitAppSubmission(yarnClient, appId, false);
+
+        // since deployment was successful, remove the hook
+        ShutdownHookUtil.removeShutdownHook(deploymentFailureHook, getClass().getSimpleName(), LOG);
+
+        return () -> waitAppSubmission(yarnClient, appId, true);
```

Comment:
What happens if this supplier is called after a long time and when the yarn application has been removed from the cluster? Won't this then fail with `"Failed to deploy the cluster."`? This seems wrong.

Reply:
Thanks for pointing this. Has distinguished between different exceptions in the latest commit.